ci/jobs/scripts/clickhouse_proc.py

```diff
@@ -620,18 +620,21 @@ def terminate(self):
             (self.proc_2, self.pid_file_replica_2, self.pid_2, self.run_path2),
         ):
             if proc and pid:
-                # Increase timeout to 10 minutes (max-tries * 2 seconds) to give gdb time to collect stack traces
-                # (if safeExit breakpoint is hit after the server's internal shutdown timeout is reached).
                 if not Shell.check(
-                    f"cd {run_path} && clickhouse stop --pid-path {Path(pid_file).parent} --max-tries 600",
+                    f"cd {run_path} && clickhouse stop --pid-path {Path(pid_file).parent} --max-tries 300 --do-not-kill",
                     verbose=True,
                 ):
-                    print("Failed to stop ClickHouse process gracefully")
+                    print(
+                        "Failed to stop ClickHouse process gracefully - send ABRT signal to generate core file"
+                    )
+                    Shell.check(f"kill -ABRT {pid}")
```

Comment:
@maxknv can you pleas also obtain stacktrace from coredump with gdb as well, and save it to some log, this will help with debugging (avoids needless core dump downloads for some cases)